sdk/monitor/monitor-opentelemetry-exporter/src/sampling.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Link, Attributes, SpanKind, Context, diag } from "@opentelemetry/api";
+import { Sampler, SamplingDecision, SamplingResult } from "@opentelemetry/sdk-trace-base";
+import { AzureMonitorSampleRate } from "./utils/constants/applicationinsights";
+
+export class ApplicationInsightsSampler implements Sampler {
+  private readonly _samplingPercentage: number;
+
+  constructor(samplingRate: number = 1) {
+    if (samplingRate > 1) {
+      diag.warn("Wrong sampling rate, data will not be sampled out");
+      samplingRate = 1;
+    }
+    this._samplingPercentage = Math.round(samplingRate * 100);
+  }
+
+  public shouldSample(
+    // @ts-ignore
+    context: Context,
+    traceId: string,
+    // @ts-ignore
+    spanName: string,
+    // @ts-ignore
+    spanKind: SpanKind,
+    attributes: Attributes,
+    // @ts-ignore
+    links: Link[]
+  ): SamplingResult {
+    let isSampledIn = false;
+    if (this._samplingPercentage == 100) {
+      isSampledIn = true;
+    } else if (this._samplingPercentage == 0) {
+      isSampledIn = false;
+    } else {
+      if (!traceId) {
+        isSampledIn = Math.random() * 100 < this._samplingPercentage;
+      } else {
+        isSampledIn = this._getSamplingHashCode(traceId) < this._samplingPercentage;
+      }
+    }
+    // Add sample rate as span attribute
+    attributes = attributes || {};
+    attributes[AzureMonitorSampleRate] = this._samplingPercentage;
+    return isSampledIn
+      ? { decision: SamplingDecision.RECORD_AND_SAMPLED, attributes: attributes }
+      : { decision: SamplingDecision.NOT_RECORD, attributes: attributes };
+  }
+
+  public toString(): string {
+    return "ApplicationInsightsSampler";
```

Comment:
The format should be ApplicationInsightsSampler{sampling_ratio} where sampling_ratio is [0,1]